### DISCUSSION TITLE
How can I have product attributes behave as product options?

### DISCUSSION THREAD

**Discussion by is-hiku** [swellstores/community#26] [Ideas]
### Use Case
For simplicity's sake, let's say I have 10 dishes under a the category Hamburger. These 10 hamburgers share a certain modifier (option) that enables the user to select from different types of breads (values) e.g. Brioche, Sourdough. Depending on the user's selection the price should change. I have been able to make this work at the product level by creating an option, creating values, adding a price to the value. It works. However, the problem is that it's not realistic for an admin to modify the price of a particular bread on EACH option for EACH product. Instead the admin should be able to go to one screen (Attributes, maybe?) update the price for a particular bread and that's it.

I think I'm lost with the concepts of product options, variants and attributes. I've read the docs, played around with the dashboard. I just can't get it to work. 

Are there any kind souls that can point me in the right direction?

**Comment by sags95** [comment]
Hey @is-hiku , that's an interesting suggestion for setting a price on product attributes. Another way to look at this would be to create variants rather than separate products (using Hamburger as the parent product). This way you can go into 1 product and change the prices for that option.

**Comment by ericingram** [comment]
Pricing by attribute would be a neat feature

**Comment by logeshswell** [comment]
@is-hiku I am moving this to Ideas section, as it seems to be an interesting feature request.
For the time being, I would suggest you to make a script or expose an internal API, which accept the attribute name and price as input and perform the following actions:
1. Fetch the list of products having the attribute name.

```node
const swell = require('swell-node').init('store-id', 'secret-key');

await swell.get('/products', {
  where: {
    'attributes.size': {
      $exists: true
    }
  }
});

```

2. Iterate through the list and make an [update variant](https://developers.swell.is/backend-api/variants/update-a-variant) API call to update the variant pricing.

Would this work for you for the time being?

**Reply by is-hiku** [reply]
@logeshswell could work, but I'm terrified of the technical debt incurred by the temporary solution. 

Currently toying with an idea. Create a new model called **modifiers** which holds an array attribute of products. Add modifiers as an attribute of products. Although, this would seem like a loop it would allow me to assign modifiers to products, which in turn contain references to other products.

I have not implemented it yet. Still pondering if this is a sensible approach. Thoughts?

**Reply by logeshswell** [reply]
@is-hiku This is a brilliant one! I would recommend you move forward with this approach.

As per your approach, you would end up modifying a single record in **modifiers** model, and it will reflect in all the corresponding products. You are trying to extend the product attributes to a one-step above the current constraints and it really sounds good to me.
